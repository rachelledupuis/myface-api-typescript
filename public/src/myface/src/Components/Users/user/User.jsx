import React from 'react'
import { Link } from 'react-router-dom'

export function User( {user} ) {
    return (
        <div>
            <Link to={`/users/${user.id}`}><p>{user.name}</p></Link>
            <img src={user.profileImageUrl} />
            <p>{user.username} {user.email}</p>
        </div>
    )
}