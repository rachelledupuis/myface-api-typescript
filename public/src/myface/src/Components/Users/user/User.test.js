import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { User } from "./User";


test("renders user's name", () => {
  // Arrange
  const user = {
    id: "1",
    name: "user name",
    profileImageUrl: "profile picture",
    username: "username",
    email: "email",
  };

  // Act
  render(<BrowserRouter><User user={user} /></BrowserRouter>
  );
  const messageElement = screen.getByText(/user name/i);

  // Assert
  expect(messageElement).toBeInTheDocument();
});

test("matches snapshot", () => {
    // Arrange
    const user = {
        id: "1",
        name: "user name",
        profileImageUrl: "profile picture",
        username: "username",
        email: "email",
      };
  
    // Act
    const tree = renderer
      .create(<BrowserRouter><User user={user} /></BrowserRouter>)
      .toJSON();
  
    // Assert
    expect(tree).toMatchSnapshot();
  });