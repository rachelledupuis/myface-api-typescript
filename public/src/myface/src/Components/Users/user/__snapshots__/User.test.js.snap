// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <a
    className="link"
    href="/users/1"
    onClick={[Function]}
  >
    <p>
      user name
    </p>
  </a>
  <img
    className="user-profile-image"
    src="profile picture"
  />
  <p>
    username
     
    email
  </p>
</div>
`;
